acceptance: TestDockerNodeJS failed

acceptance.TestDockerNodeJS [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/12235740?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/12235740?buildTab=artifacts#/) on release-23.2 @ [0ceb71028e018afa6855615f4cfba613eab1eb74](https://github.com/cockroachdb/cockroach/commits/0ceb71028e018afa6855615f4cfba613eab1eb74):


```
=== RUN   TestDockerNodeJS
    test_log_scope.go:170: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/_tmp/882c216ad52e28515324b9305d5301f5/logTestDockerNodeJS333170885
    test_log_scope.go:81: use -show-logs to present logs inline
    util_docker.go:62: non-zero exit code: 1
    adapter_test.go:87: -- test log scope end --
test logs left over in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/_tmp/882c216ad52e28515324b9305d5301f5/logTestDockerNodeJS333170885
--- FAIL: TestDockerNodeJS (65.53s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112266 acceptance: TestDockerNodeJS failed [C-test-failure O-robot T-sql-foundations branch-master release-blocker]
- #111968 acceptance: TestDockerNodeJS failed [NotLeaseHolderErrors are logged before node is done starting up] [C-test-failure O-robot T-kv X-noreuse branch-master release-blocker]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDockerNodeJS.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-32514

--- a/pkg/acceptance/cluster/dockercluster.go
+++ b/pkg/acceptance/cluster/dockercluster.go
@@ -461,9 +461,13 @@ func (l *DockerCluster) createNodeCerts() {
 
 // startNode starts a Docker container to run testNode. It may be called in
 // parallel to start many nodes at once, and thus should remain threadsafe.
-func (l *DockerCluster) startNode(ctx context.Context, node *testNode) {
+func (l *DockerCluster) startNode(ctx context.Context, node *testNode, singleNode bool) {
+	startCmd := "start"
+	if singleNode {
+		startCmd = "start-single-node"
+	}
 	cmd := []string{
-		"start",
+		startCmd,
 		"--certs-dir=/certs/",
 		"--listen-addr=" + node.nodeStr,
 		"--vmodule=*=1",
@@ -483,8 +487,10 @@ func (l *DockerCluster) startNode(ctx context.Context, node *testNode) {
 		cmd = append(cmd, fmt.Sprintf("--store=%s", storeSpec))
 	}
 	// Append --join flag for all nodes.
-	firstNodeAddr := l.Nodes[0].nodeStr
-	cmd = append(cmd, "--join="+net.JoinHostPort(firstNodeAddr, base.DefaultPort))
+	if !singleNode {
+		firstNodeAddr := l.Nodes[0].nodeStr
+		cmd = append(cmd, "--join="+net.JoinHostPort(firstNodeAddr, base.DefaultPort))
+	}
 
 	dockerLogDir := "/logs/" + node.nodeStr
 	localLogDir := filepath.Join(l.volumesDir, "logs", node.nodeStr)
@@ -499,6 +505,9 @@ func (l *DockerCluster) startNode(ctx context.Context, node *testNode) {
 		// Needed for backward-compat on crdb_internal.ranges{_no_leases}.
 		// Remove in v23.2.
 		"COCKROACH_FORCE_DEPRECATED_SHOW_RANGE_BEHAVIOR=false",
+		// Disable metamorphic testing for acceptance tests, since they are
+		// end-to-end tests and metamorphic constants can make them too slow.
+		"COCKROACH_INTERNAL_DISABLE_METAMORPHIC_TESTING=true",
 	}
 	l.createRoach(ctx, node, l.vols, env, cmd...)
 	maybePanic(node.Start(ctx))
@@ -663,15 +672,16 @@ func (l *DockerCluster) Start(ctx context.Context) {
 	go l.monitor(ctx, l.monitorDone)
 	var wg sync.WaitGroup
 	wg.Add(len(l.Nodes))
+	singleNode := len(l.Nodes) == 1
 	for _, node := range l.Nodes {
 		go func(node *testNode) {
 			defer wg.Done()
-			l.startNode(ctx, node)
+			l.startNode(ctx, node, singleNode)
 		}(node)
 	}
 	wg.Wait()
 
-	if l.config.InitMode == INIT_COMMAND && len(l.Nodes) > 0 {
+	if l.config.InitMode == INIT_COMMAND && len(l.Nodes) > 1 {
 		l.RunInitCommand(ctx, 0)
 	}
 }

--- a/pkg/acceptance/util_docker.go
+++ b/pkg/acceptance/util_docker.go
@@ -208,6 +208,9 @@ var cmdBase = []string{
 	"/usr/bin/env",
 	"COCKROACH_SKIP_UPDATE_CHECK=1",
 	"COCKROACH_CRASH_REPORTS=",
+	// Disable metamorphic testing for acceptance tests, since they are
+	// end-to-end tests and metamorphic constants can make them too slow.
+	"COCKROACH_INTERNAL_DISABLE_METAMORPHIC_TESTING=true",
 	"/bin/bash",
 	"-c",
 }

